### PR TITLE
Continuation externalising cmdline

### PR DESCRIPTION
**Pull request by Akin909**
@bryphe @Bretley  following my comment in #1127, I've cleaned up the merge conflicts, so we can bring in the changes you made to externalize the `cmdline`, 

just to note that the primary ongoing issue I can see is the lack of a *proper* cursor, at the moment I've tweaked the component to be an input and force focus so it seems like its cursor responds to input but actually this is controlled by redux so the cursor doesn't behave properly.

EDIT: @Bretley I can also make this PR to your branch if you'd prefer to merge it in then we merge in the original.

### CODE REVIEW

File: browser/src/UI/components/CommandLine.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    }
+`
+
+const CommandLineBox = styled.div`
+    position: relative;
+    margin-top: 16px;
+    padding: 8px;
+    width: 75%;
+    max-width: 900px;
+    background-color: ${p => p.theme.background};
```
Comment (bryphe) [inline browser/src/UI/components/CommandLine.tsx]: Cool, seems like this what you were mentioning - I need to try this out in #1159 !

**Comment by bryphe** [comment]
I'm good to bring this in - as you mentioned I don't want it to get lost in a sea of merge conflicts. Thanks for bringing it up-to-date @Akin909 , and thanks @samvv and @Bretley for your work on this! I'll bring this in now and merge #1175 with it tomorrow.

**Comment by Bretley** [comment]
Sorry I've been AWOL since finals and visiting family, things have gotten busy as there's been an unexpected death recently in my family. I've got no objections to you fixing my initial work up; I'm not going to continue it. Keep up the great work folks

**Comment by bryphe** [comment]
> Sorry I've been AWOL since finals and visiting family, things have gotten busy as there's been an unexpected death recently in my family.

Sorry to hear that 😦 Hope everything is okay!

>  I've got no objections to you fixing my initial work up; I'm not going to continue it.

Thanks for getting us started here!

**Pull request merged**